backend/routes/userRoutes.js

```diff
@@ -210,12 +240,30 @@ router.get('/recommendedEvents/:userId', async (req, res) => {
       distanceScore = 1
     }
     // -------------------------------------------------------------------------------------------------------------------------
+    // Comments Score -----------------------------------------------------------------------------------------------------------
+    countOfCommentsInEvent = 0
+    for (let i = 0; i < totalComments; i++) {
+      if (idEventsCommented[i] == e.id){
+        countOfCommentsInEvent = countOfCommentsInEvent + 1
+      }
+    }
+    if (totalComments != 0) {
+      commentScore = (countOfCommentsInEvent)/(totalComments)
+    } else {
+      commentScore = 0
+    }
+    // ------------------------------------------------------------------------------------------------------------------------
+    // Popularity Score -----------------------------------------------------------------------------------------------------------
+    popularityScore = 0
+    if (activeUsers.length != 0) popularityScore = eventAttendance/(activeUsers.length)
+    // ------------------------------------------------------------------------------------------------------------------------
     // Total Score -----------------------------------------------------------------------------------------------------------
-    totalScore = dateScore + categoryScore + distanceScore
+    totalScore = dateScore + categoryScore + distanceScore + commentScore + 4*popularityScore
     // -------------------------------------------------------------------------------------------------------------------------
-
-    idRecommendedEvents.push(e.id)
-    scoreRecommendedEvents.push(totalScore)
+    if (!idEventsAttending.includes(e.id)){
+      idRecommendedEvents.push(e.id)
+      scoreRecommendedEvents.push(totalScore)
+    }
```

Comment:
Consider moving all the event filtering to the start of the recommendation logic so that we don't need to calculate scores for the events that we know we don't need to recommend.

Reply:
Done